app/routes/__auth/user.server.ts

```diff
@@ -14,8 +16,59 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
+  const token = session.get('token') as string
   if (!sub) return null
+  const user = { sub, email, groups, idp, refreshToken, cognitoUserName, token }
+  if (
+    !token ||
+    JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp *
+      1000 <
+      Date.now()
+  ) {
+    await refreshUser(user)
+  }
+  return user
+}
+
+export async function updateSession(user: any) {
+  const session = await storage.getSession()
+  Object.entries(user).forEach(([key, value]) => {
+    session.set(key, value)
+  })
+  await storage.commitSession(session)
+}
+
+export async function clearUserToken(sub: string) {
+  const db = await tables()
+  const targetSessionResults = await db.sessions.query({
+    IndexName: 'bySub',
+    KeyConditionExpression: '#sub = :sub',
+    ExpressionAttributeNames: {
+      '#sub': 'sub',
+      '#idx': '_idx',
+    },
+    ExpressionAttributeValues: {
+      ':sub': sub,
+    },
+    ProjectionExpression: '#idx',
+  })
+
+  if (!targetSessionResults.Items) return
+
+  const promises = targetSessionResults.Items.map((item) =>
+    db.sessions.update({
+      Key: { _idx: item['_idx'] as string },
+      UpdateExpression: 'set #token = :nullToken',
```

Comment:
Just delete the access token. (Don't just set it to null.)

Reply:
I didnt realize there was a command to just REMOVE an attribute, trying that now